src/metatensor/models/cli/train_model.py

```diff
@@ -88,17 +96,63 @@ def train_model(options: DictConfig) -> None:
     https://hydra.cc/docs/advanced/hydra-command-line-flags/ and
     https://hydra.cc/docs/advanced/override_grammar/basic/ for details.
 
+    :param options: Options file path
+    :param output: Path to save the final model
+    :param hydra_parameters: Hydra's command line and override flags
+    """
+    conf = OmegaConf.load(options)
+
+    try:
+        architecture_name = conf["architecture"]["name"]
+    except ConfigKeyError as exc:
+        raise ConfigKeyError("Architecture name is not defined!") from exc
+
+    conf["defaults"] = [
+        "base",
+        {"architecture": architecture_name},
+        {"override hydra/job_logging": "custom"},
+        "_self_",
+    ]
+
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        options_new = Path(tmpdirname) / "options.yaml"
+        OmegaConf.save(config=conf, f=options_new)
+
+        # HACK: Hydra parses command line arguments directlty from `sys.argv`. We
+        # override `sys.argv` to be compatible with our CLI architecture.
+        argv = sys.argv[:1]
+
+        argv.append(f"--config-dir={options_new.parent}")
+        argv.append(f"--config-name={options_new.name}")
+        argv.append(f"+output_path={output}")
+
+        if hydra_parameters is not None:
+            argv += hydra_parameters
+
+        sys.argv = argv
+
+        _train_model_hydra()
+
+
+@hydra.main(config_path=str(CONFIG_PATH), version_base=None)
+def _train_model_hydra(options: DictConfig) -> None:
+    """Actual fit function called in :func:`train_model`.
+
     :param options: A dictionary-like object obtained from Hydra, containing all the
         necessary options for dataset preparation, model hyperparameters, and training.
     """
+    if options["base_precision"] == 64:
+        torch.set_default_dtype(torch.float64)
+    elif options["base_precision"] == 32:
+        torch.set_default_dtype(torch.float32)
+    elif options["base_precision"] == 16:
+        torch.set_default_dtype(torch.float16)
+    else:
+        raise ValueError("Only 64, 32 or 16 are possible values for `base_precision`.")
 
-    # This gives some accuracy improvements. It is very likely that
-    # this is just due to the preliminary composition fit in the SOAP-BPNN.
-    # TODO: investigate
-    torch.set_default_dtype(torch.float64)
-
-    # TODO load seed from config
     generator = torch.Generator()
```

Comment:
As far as I know, this is not a global generator, meaning that you can feed it to random functions to control their behavior. I think we should have a set_random_seed function in the utils that does at least
```
torch.manual_seed(seed)  # this is global
np.random.seed(seed)  # we use NumPy to shuffle the datasets IIRC
random.seed(seed)  # just to make sure
```
And then we can ask Sergey if he wants to improve it (although in my experience the above has always worked well).

Reply:
Yes, I didn't wanted to check this i this PR. We can do it in an upcoming one.